Source/CarthageKitTests/Fakes/FakeURLSession.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+final class FakeDownloadTask: NSURLSessionDownloadTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeDataTask: NSURLSessionDataTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeURLSession: NSURLSession {
```

Comment:
Is `NSURLSession` meant to be subclassed? I don't think so... so this is probably fragile and relying on `NSURLSession`'s implementation details (_insert note about `final` default here_).

Instead of mocking requests like this, I'd recommend creating a `protocol URLSessionType` that `NSURLSession` conforms to, and so that `FakeURLSession` does **not** inherit `NSURLSession`, but instead implements all the fake logic by itself.